tools/astarte_dev_tool/lib/constants/system.ex

```diff
@@ -0,0 +1,31 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteDevTool.Constants.System do
+  @command "docker"
+  @command_up_args ~w(compose -f docker-compose.yml -f docker-compose.dev.yml up --build --watch -d)
+  @command_down_args ~w(compose -f docker-compose.yml -f docker-compose.dev.yml down)
+  @command_watch_args ~w(compose -f docker-compose.yml -f docker-compose.dev.yml watch --no-up)
+  @options [stderr_to_stdout: true, into: IO.stream(:stdio, :line)]
+
+  def command, do: @command
+  def command_up_args, do: @command_up_args
+  def command_down_args, do: @command_down_args
+  def command_watch_args, do: @command_watch_args
+  def options, do: @options
```

Comment:
Since `Constants.options() ++ [cd: path]` is in every call to `Constants.options()`, could this be parametrized? Or maybe add some keyword options, like
```elixir
def options(opts \\ []) do
  if Keyword.get(opts, :cd)  do
  #etc etc
  end
end

Reply:
I'd rather not: in this case `options` is always used with functionality that requires the `path` but it won't necessarily be so. It's related to _any_ bash command

Reply:
Using keyword options would allow to write both these calls:
- `Constants.options(cd: path)`
- `Constant.options()`
And also allow for extensibility of the options list

Reply:
Sorry, but I don't get it, if we parameterize  eventual extra options to the Constants.options call, we get something like
```
System.cmd(Constants.command(), args, Constants.options( [cd: path])) 
```

instead of 

```
System.cmd(Constants.command(), args, Constants.options() ++ [cd: path])
```

Imho it will end up cleaner, and every optional params can still be used

Reply:
I really don't like to assume that constants are functions, so I would have preferred a metaprogramming approach (https://medium.com/@chakravartyaashish/creating-constant-modules-in-elixir-497590a5283f). Tied to that, I want to keep the constants as agnostic as possible

Reply:
I've also read this https://dev.to/olddutchcap/two-kinds-of-elixir-constants-5gad

Reply:
I'd agree with @shinnokdisengir here, the options passed in `@options` (which I would call `base_opts` since it better conveys the idead) are universal and I'd expect to use them in every command, while the fact that you have to `cd` is very specific to this set of commands. Passing `cd` to the function with a single data point seems like premature refactoring, so I'd stay with `Constants.base_opts() ++ [cd: path]` for the time being.

Reply:
@rbino I'm not sure of that as keyword lists would allow to add a number of other options in the future, too. Let's keep the option (pun intented) in mind when developing next and move on, however.